aggregators/1.0/modules/ROOT/pages/aggregators-size-example.adoc

```diff
@@ -0,0 +1,141 @@
+= Size-Based Aggregator Example - Mule 4
+
+The Aggregators module *Size based aggregator* scope enables you to aggregate elements until a pre-defined size number of elements completes the aggregation. The following example shows how to configure this scope to aggregate up to three HTTP request calls. Then transform the calls via DataWeave into random ID numbers. When the aggregation reaches the three calls, the application logs all the aggregated random ID number elements.
+
+.Size-Based Aggregator Flow Example
+image::aggregator-size-example.png[Size-Based Aggregator Flow Example in Anypoint Studio Canvas]
+
+To accomplish this example, you must create the Mule application and test it with curl commands.
+
+
+== Create the Mule Application
+
+To create the Mule flow in Anypoint Studio:
+
+. In *Mule Palette*, select *HTTP Listener* and drag it on to the canvas. +
+The source initiates the flow by listening for incoming HTTP message attributes.
+. In *Connector configuration*, select *HTTP_Listener_config* global configuration.
+. Set *Path* to `/test`.
+. Drag a *Transform message* component to the right of *Listener*.
+. In the DataSense preview window of the *Transform message*, add the following DataWeave expression that transforms each HTTP request by assigning a `Source Name` as `"size-based"` and a `Source ID` as `random() as String`:
++
+[source,DataWeave,linenums]
+----
+%dw 2.0
+output application/json
+---
+{
+    "Source Name": "size-based",
+    "Source ID": (random() as String)
+}
+----
+[start=6]
+. Drag the *Size based aggregator* scope to the right of *Transform message*.
+. Set *Max size* to `3` which is the total number of elements to be aggregated before considering the aggregation complete.
+. Set *Content* to `payload` which is the expression that defines what to aggregate, in this case the transformed HTTP requests.
+. Drag a *Logger* component into the *Incremental aggregation* route.
+. In the *Logger* configuration screen, set *Message* to `Doing incremental aggregation step.` +
+. Drag a *Logger* component into the *Aggregation complete* route.
+. Set *Message* to `Aggregation complete`.
+. Drag another *Logger* to the right of the first *Logger* in *Aggregation complete*.
+. Set *Message* to the following expression that returns all the aggregated elements:
++
+[source,xml,linenums]
+----
+output application/json
+---
+payload
+----
+. Save the project.
+. In *Package Explorer* click *Run > Run As > Mule Application*.
+. Test the app by sending three times the curl command `http://localhost:8086/test`. +
+This generates three HTTP requests for the *Size based aggregator* to complete.
```

Comment:
Should this say "to process," since the whole thing is not complete until all three are addressed?